test/units/utils/display/test_warning.py

```diff
@@ -27,7 +27,7 @@ def test_warning(capsys, mocker, warning_message):
     d.warning(warning_message)
     out, err = capsys.readouterr()
     assert d._warns == {expected_warning_message: 1}
-    assert err == '\x1b[1;35m{0}\x1b[0m\n\x1b[1;35m\x1b[0m'.format(expected_warning_message.rstrip('\n'))
+    assert err == '\x1b[1;35m{0}\x1b[0m\n'.format(expected_warning_message.rstrip('\n'))
```

Comment:
This was originally looking for:
`\x1b[1;35m` (color on) - `{0}` (the message) - `\x1b[0m` (color off) - `\n` (newline) - `\x1b[1;35m` (color on) - `\x1b[0m` (color off)

With this PR it now puts the newline _outside_ the coloration, and does not unnecessarily turn the coloration on and off a second time.  So it is now looking for:
`\x1b[1;35m` (color on) - `{0}` (the message) - `\x1b[0m` (color off) - `\n` (newline)